packages/bundler/src/modules/BundleManager.ts

```diff
@@ -131,19 +131,40 @@ export class BundleManager {
       } = parsedError.args
       const userOp = userOps[opIndex]
       const reasonStr: string = reason.toString()
+
+      // find entity to blame
+      let addr: string | undefined
       if (reasonStr.startsWith('AA3')) {
-        this.reputationManager.crashedHandleOps(userOp.paymaster)
+        // [EREP-030] A staked account is accountable for failure in any entity
+        addr = await this.isAccountStaked(userOp) ? userOp.sender : userOp.paymaster
       } else if (reasonStr.startsWith('AA2')) {
-        this.reputationManager.crashedHandleOps(userOp.sender)
+        // [EREP-020] A staked factory is "accountable" for account
+        addr = await this.isFactoryStaked(userOp) ? userOp.factory : userOp.sender
       } else if (reasonStr.startsWith('AA1')) {
-        this.reputationManager.crashedHandleOps(userOp.factory)
+        // [EREP-030] A staked account is accountable for failure in any entity
+        addr = await this.isAccountStaked(userOp) ? userOp.sender : userOp.factory
```

Comment:
I think you told be this situation is impossible. `AA1x` indicates an error in the Factory contract during validation, and `isAccountStaked` cannot be true for an account currently being created. Am I missing something?